bbq/vm/vm.go

```diff
@@ -851,33 +829,20 @@ func getStringConstant(vm *VM, index uint16) string {
 func opTransfer(vm *VM, ins opcode.InstructionTransfer) {
 	typeIndex := ins.Type
 	targetType := vm.loadType(typeIndex)
-	value := vm.peek()
 
-	config := vm.context
+	context := vm.context
 
-	transferredValue := value.Transfer(
-		config,
-		EmptyLocationRange,
-		atree.Address{},
-		false,
-		nil,
-		nil,
+	value := vm.peek()
+	valueType := value.StaticType(context)
 
-		// TODO: Pass the correct flag here
-		false,
+	transferredValue := interpreter.TransferAndConvert(
```

Comment:
Updated to match the interpreter: Transfer should not only transfer, but also should convert if necessary.